docs/jumpstart/week-3.md

```diff
@@ -0,0 +1,418 @@
+    list_of_values: [1, 'five', 4.0, "six"],
+    aNestedObject: {
+        innerKey: "outer key"
+    },
+    oneLastValue: "Notice this doesn't have a comma"
+};
+```
+
+Properties have a colon separating the key from the value and they end with a comma if there are more properties in the object.
+
+#### Operations
+
+You refer to properties in an object by their key. You can either use dot notation `someObject.someKey` or array like notation `someObject['someKey']`. Dot notation is shorter but array like notation gives you the flexibility to use a variable as the key. If you try to access a property that doesn't exist, you'll get the undefined value.
+
+You can add properties to an object just with simple assignment statements, similar to updating an index in an array.
+
+```js
+const person = {
+    name: "Mike",
+    age: 12
+};
+const personName = person.name; // Equal to Mike
+const personAge = person['age']; // Equal to 12
+const personWeight = person.weight; // Equal to undefined
+
+const weightProperty = 'weight';
+person[weightProperty] = 150;
+const newPersonWeightOne = person[weightProperty] // Equal to 150
+const newPersonWeightTwo = person.weight; // Equal to 150
+```
+
+
+### null and undefined
+
+`null` and `undefined` are two additional primitive types that Javascript defines. While they will come up, it's almost never a good idea to intentionally use them in your code design.
+
+`null` refers to an unintialized object. This is not the same thing as an empty object (`{}`).
+
+`undefined` is defined as an unitialized variable.
+
+We will see a couple times when these values may come up, but generally if they appear where you do not expect them it's a good indication that you have a bug in your code.
+
+
+## Control Flow
+
+Control flow allows you to execute different pieces of code based on different conditions.
+
+A condition is a statement that evaluates to a *boolean* value (`true` or `false`).
+
+### if & else
+
+`if` statements are the most basic form of control flow. They take a single condition and execute a given block of code once if the condition is true. This can also be combined with an `else` statement that will execute a block of code if the condition is false.
+
+```js
+const conditionOne = false;
+const conditionTwo = true;
+
+if (conditionOne) {
+    // Code block 1
+} else if (conditionTwo) {
+    // Code block 2
+} else {
+    // Code block 3
+}
+```
+
+### while
+
+`while` statements allow you to execute a block of code multiple times based on a changing condition. They take a single condition and execute the given code block until the condition evaluates to false.
+
+```js
+let counter = 0;
+while (counter < 4) {
+    console.log(counter);
+    counter = counter + 1;
+}
+```
+
+You have to make sure that the condition you use in a while loop will change over execution. If you use `while (true) {...}` for example, you will create an infinite loop and your program will likely crash after running out of memory.
```

Comment:
Introduce `break;` and `continue;`? I know you show early return down below. I think this might be a bit extreme since there are definitely cases for `while(true)`